### PR TITLE
Remove nelmio v4 from conflict

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this change should be done here.

## Changelog

```markdown
### Removed
- Removed conflict with `nelmio/api-doc-bundle` v4
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
     "suggest": {
         "friendsofsymfony/rest-bundle": "For using the public API methods.",
         "jms/serializer": "For using the public API methods.",
-        "nelmio/api-doc-bundle": "For using the public API methods.",
+        "nelmio/api-doc-bundle": "v2 or v3 - for using the public API methods. v4 does not provide API.",
```
Comment (phansys) [inline composer.json]:
```suggestion
        "nelmio/api-doc-bundle": "Versions 2.x or 3.x for using the public ReST API methods. Version 4.x is allowed but not supported.",
```

File: src/DependencyInjection/SonataUserExtension.php [hunk 2]
```diff
@@ -88,7 +89,9 @@ class_exists('Sonata\GoogleAuthenticator\GoogleAuthenticator')) {
             $loader->load('serializer.xml');
 
             $loader->load('api_form.xml');
-            if (class_exists(Operation::class)) {
+            if (class_exists(Operation::class) && class_exists(OpenApiOperation::class)) {
+                // UserBundle does not provide API for nelmio v4. This code allow remove nelmio v4 from conflict.
```
Comment (phansys) [inline src/DependencyInjection/SonataUserExtension.php]:
```suggestion
                // UserBundle does not support "nelmio/api-doc-bundle" 4.x and does not provide an API through this version. This condition allows to accept setups using "nelmio/api-doc-bundle" 4.x dependency without conflicts.
                // @no-op.
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Comment by wbloszyk** [comment]
In cases when people using API and havent defind nelmio version in own composer then v4 will be installed. Aby idea haw resole it?
@phansys @sonata-project/contributors

**Comment by phansys** [comment]
>In cases when people using API and havent defind nelmio version in own composer then v4 will be installed. Aby idea haw resole it?

Why the package will be installed in that case? IIUC, in your example the package is not required by the userland project, and neither is it required by "sonata-project/user-bundle".
Or do you mean the userland project requiring something like `"sonata-project/user-bundle": "*"`?
If it is the case, I think is responsibility of the user to get rid about the resolved dependencies when performing an update, as the problem there is the loose constraint.

**Review by VincentLanglet — APPROVED** [review]

**Review by phansys — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**